src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx

```diff
@@ -116,34 +115,27 @@ export default class OrganizationDiscover extends React.Component {
       // This indicates navigation changes (e.g. back button on browser)
       // We need to update our search store and probably runQuery
       const {projects, range, start, end, utc} = newQuery;
-      let hasChange = false;
 
       if (projects) {
         this.updateProjects(projects);
-        hasChange = true;
       }
 
-      if (range || (end && start)) {
```

Comment:
this is no longer true, null are now valid values for all of range/end/start. Because of this `hasChanges` is no longer relevant, (I'm not sure if it was needed since we check if `search` has changed earlier in this lifecycle method).

Reply:
I think it detects if there are changes when a saved query is being edited and this is no longer happening